packages/truffle-box/lib/utils/unbox.js

```diff
@@ -1,88 +1,73 @@
-        .replace(/#.*/, "") + "/master/truffle-box.json"
-    );
-
-    var options = {
-      method: "HEAD",
-      uri: "https://" + configURL.host + configURL.path
-    };
-    request(options, function(error, r) {
-      if (error) {
-        return reject(
-          new Error(
-            "Error making request to " +
-              options.uri +
-              ". Got error: " +
-              error.message +
-              ". Please check the format of the requested resource."
-          )
-        );
-      } else if (r.statusCode === 404) {
-        return reject(
-          new Error(
-            "Truffle Box at URL " +
-              url +
-              " doesn't exist. If you believe this is an error, please contact Truffle support."
-          )
-        );
-      } else if (r.statusCode !== 200) {
-        return reject(
-          new Error(
-            "Error connecting to github.com. Please check your internet connection and try again."
-          )
-        );
-      }
-      accept();
-    });
-  });
+  const configURL = parseURL(
+    `${vcsurl(url)
+      .replace("github.com", "raw.githubusercontent.com")
+      .replace(/#.*/, "")}/master/truffle-box.json`
+  );
+
+  const options = {
+    method: "HEAD",
+    uri: `https://${configURL.host}${configURL.path}`,
+    resolveWithFullResponse: true,
+    simple: false
+  };
+
+  try {
+    const { statusCode } = await rp(options);
+    if (statusCode === 404) {
+      throw new Error(
+        `Truffle Box at URL ${url} doesn't exist. If you believe this is an error, please contact Truffle support.`
+      );
+    } else if (statusCode !== 200) {
+      throw new Error(
+        "Error connecting to github.com. Please check your internet connection and try again."
+      );
+    }
+  } catch (error) {
+    throw error;
+  }
 }
 
 function fetchRepository(url, dir) {
-  return new Promise(function(accept, reject) {
+  return new Promise((accept, reject) => {
     // Download the package from github.
     ghdownload(url, dir)
-      .on("err", function(err) {
+      .on("err", err => {
         reject(err);
       })
-      .on("end", function() {
+      .on("end", () => {
         accept();
       });
   });
 }
 
-function prepareToCopyFiles(tempDir, boxConfig) {
-  const needingRemoval = boxConfig.ignore || [];
+function prepareToCopyFiles(tempDir, { ignore }) {
+  const needingRemoval = ignore;
 
   // remove box config file
   needingRemoval.push("truffle-box.json");
   needingRemoval.push("truffle-init.json");
 
   const promises = needingRemoval
     .map(fileName => path.join(tempDir, fileName))
-    .map(
-      filePath =>
-        new Promise((resolve, reject) => {
-          fs.remove(filePath, error => {
-            if (error) return reject(error);
-            resolve();
-          });
-        })
-    );
+    .map(async filePath => {
+      try {
+        fse.removeSync(filePath);
+      } catch (error) {
+        throw error;
+      }
+    });
```

Comment:
Hm we don't actually use those promises for anything... we can just change this to a single `.map` and an outer `for` loop